tests/countries/test_finland.py

```diff
@@ -165,6 +167,152 @@ def test_all_saints_day(self):
             "1957-11-01",
         )
 
+    def _test_unofficial_holiday(self, name, month_and_day, year_of_adoption):
+        self.assertHolidayName(
+            name,
+            self.unofficial_holidays,
+            (f"{year}-{month_and_day}" for year in range(year_of_adoption, 2031)),
+        )
+        self.assertNoHolidayName(name, self.unofficial_holidays, year_of_adoption - 1)
+
+    def test_runeberg_day(self):
+        self._test_unofficial_holiday("Runebergin päivä", "02-05", 1854)
```

Comment:
And after refactoring you would be able to use it this way:
```suggestion
        self._test_unofficial_holiday("Runebergin päivä", "1854-02-05")
```